src/lightning/pytorch/callbacks/finetuning.py

```diff
@@ -92,11 +92,10 @@ def state_dict(self) -> Dict[str, Any]:
 
     def load_state_dict(self, state_dict: Dict[str, Any]) -> None:
         self._restarting = True
-        if "internal_optimizer_metadata" in state_dict:
-            self._internal_optimizer_metadata = state_dict["internal_optimizer_metadata"]
-        else:
-            # compatibility to load from old checkpoints before PR #11887
-            self._internal_optimizer_metadata = state_dict  # type: ignore[assignment]
+        # compatibility to load from old checkpoints before PR #11887
+        self._internal_optimizer_metadata = (  # type: ignore[assignment]
```

Comment:
Same suggestion

Reply:
if you want to keep the long version, we can `noqa` it...

Reply:
reverted